docs/kubernetes/windows.md

```diff
@@ -1,446 +1,358 @@
+4. Set the Azure service principal for the deployments
 
-#### 3. Create a pod with azure disk pvc
-```kubectl apply -f https://raw.githubusercontent.com/JiangtianLi/Examples/master/windows/azuredisk/iis-azuredisk.yaml```
+You can use the same PowerShell window from earlier to run this next script to do all that for you. Be sure to replace `$dnsPrefix` with something unique and descriptive, `$windowsUser` and `$windowsPassword` to meet the requirements.
 
-#### watch the status of pod until its `STATUS` is `Running`
-```
-watch kubectl get po/iis-azuredisk -o wide
-```
+```powershell
+# Be sure to change these next 3 lines for your deployment
+$dnsPrefix = "wink8s1"
+$windowsUser = "winuser"
+$windowsPassword = "Cr4shOverride!"
 
-#### 4. Enter the pod container to validate
-```
-kubectl exec -it iis-azuredisk -- cmd
-```
+# Load template
+$inJson = Get-Content .\kubernetes-windows.json | ConvertFrom-Json
 
+# Set dnsPrefix
+$inJson.properties.masterProfile.dnsPrefix = $dnsPrefix
 
-## Example using multiple containers in a POD
-```yaml
-apiVersion: extensions/v1beta1
-kind: Deployment
-metadata:
-  labels:
-    app: two-containers
-  name: two-containers
-spec:
-  replicas: 1
-  template:
-    metadata:
-      labels:
-        app: two-containers
-      name: two-containers
-    spec:
-      volumes:
-      - name: shared-data
-        emptyDir: {}
+# Set Windows username & password
+$inJson.properties.windowsProfile.adminPassword = $windowsUser
+$inJson.properties.windowsProfile.adminUsername = $windowsPassword
 
-      containers:
+# Copy in your SSH public key from `~/.ssh/id_rsa.pub` to linuxProfile.ssh.publicKeys.keyData
+$inJson.properties.linuxProfile.ssh.publicKeys[0].keyData = [string](Get-Content "~/.ssh/id_rsa.pub")
 
-        - name: iis-container
-          image: microsoft/iis:windowsservercore-1803
-          volumeMounts:
-          - name: shared-data
-            mountPath: /wwwcache
-          command: 
-          - powershell.exe
-          - -command 
-          - "while ($true) { Start-Sleep -Seconds 10; Copy-Item -Path C:\\wwwcache\\iisstart.htm -Destination C:\\inetpub\\wwwroot\\iisstart.htm; }"            
-
-        - name: servercore-container
-          image: microsoft/windowsservercore:1803
-          volumeMounts:
-          - name: shared-data
-            mountPath: /poddata
-          command: 
-          - powershell.exe
-          - -command 
-          - "$i=0; while ($true) { Start-Sleep -Seconds 10; $msg = 'Hello from the servercore container, count is {0}' -f $i; Set-Content -Path C:\\poddata\\iisstart.htm -Value $msg; $i++; }"
+# Set servicePrincipalProfile
+$inJson.properties.servicePrincipalProfile.clientId = $sp.appId
+$inJson.properties.servicePrincipalProfile.secret = $sp.password
 
-      nodeSelector:
-        beta.kubernetes.io/os: windows
+# Save file
+$inJson | ConvertTo-Json -Depth 5 | Out-File -Encoding ascii -FilePath "kubernetes-windows-complete.json"

```

Comment:
I would add in commands, again, for *nix and macOS. A lot of the JSON conversion is unnecessary when doing this same template generation from mac and linux